File: backend/store/app/domain/data/customer.go
package data

type Customer struct {
	Id              EntityId `gorm:"primaryKey"`
	Name            string
	Phone           string
	DeliveryAddress string
}

func (c *Customer) GetId() EntityId {
	return c.Id
}

func (c *Customer) SetId(id EntityId) {
	c.Id = id
}

File: ecommerce/src/main/java/com/bookstore/ecommerce/app/domain/Customer.java
package com.bookstore.ecommerce.app.domain;

import java.util.UUID;
import com.google.common.base.Strings;
import lombok.Getter;
import lombok.var;

public class Customer {
  @Getter
  private com.bookstore.ecommerce.app.domain.data.Customer customer;

  public Customer(com.bookstore.ecommerce.app.domain.data.Customer customer) {
    var cloned = customer.clone();
    if (Strings.isNullOrEmpty(cloned.getId())) {
      cloned.setId(UUID.randomUUID().toString());
    }

    this.customer = cloned;
  }

  public com.bookstore.ecommerce.app.domain.data.Customer getState() {
    return this.customer.clone();
  }
}
